### PR TITLE
add js chaincode for the private data concept

### PR DESCRIPTION
**Pull request by abdou-chakhkhar**
Signed-off-by: abdou.chakhkhar <abdelmoula.chakhkhar@uit.ac.ma>

In this PR, I implemented the js chaincode for the private data concept, which s gonna allow the js developers to get familiarized so quickly with private data APIs in the nodejs SDK.

### CODE REVIEW

File: asset-transfer-private-data/chaincode-javascript/package.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "lint": "eslint *.js */**.js",
+        "pretest": "npm run lint",
+        "test": "nyc mocha --recursive",
+        "start": "fabric-chaincode-node start"
+    },
+    "engineStrict": true,
+    "author": "Hyperledger",
+    "license": "Apache-2.0",
+    "dependencies": {
+        "fabric-contract-api": "^2.0.0",
```
Comment (mbwhite) [inline asset-transfer-private-data/chaincode-javascript/package.json:20]: would you be able to make the fabric dependencies the latest please.

File: asset-transfer-private-data/chaincode-javascript/package.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        "fabric-shim": "^2.0.0",
+        "json-stringify-deterministic": "^1.0.1",
+        "node-forge": "^1.3.1",
+        "openssl-nodejs": "^1.0.5",
```
Comment (mbwhite) [inline asset-transfer-private-data/chaincode-javascript/package.json:24]: what are node-forge and openssl-nodejs being used for?

**Review by mbwhite — COMMENTED** [review]
Thank you for submitting  PR.. a couple of comments.

If you are able to, some formatting as gone astray, and some commented out code it's best to remove. If it's useful for reference please add a comment saying why.

thanks!

**Comment by lehors** [comment]
@abdou-chakhkhar Could you please respond to @mbwhite? Thanks.

**Comment by lehors** [comment]
Not having received any response I'm closing this. We unfortunately cannot afford to take on new code.
If you do want to pick it up and address the comments, please, do reopen it.
Thank you for your contribution.

**Pull request closed without merging**